R/chat.R

```diff
@@ -388,6 +392,23 @@ Chat <- R6::R6Class(
       invisible(self)
     },
 
+    #' @description Register callbacks for specific events. For expert use only.
+    #'
+    #' @param event The name of the event. Supported events include
+    #'   `tool_request`, and `tool_result`.
+    #' @param callback A function to be called when the event occurs. The
+    #'   function must take at least one argument, which is the data associated
+    #'   with the event:
+    #'
+    #'   * `tool_request` receives a `ContentToolRequest` object.
+    #'   * `tool_result` receives a `ContentToolResult` object.
+    #'
+    #' @return Returns a function that can be called to remove the callback.
+    register_callback = function(event, callback) {
```

Comment:
Even here, I think there's some advantage to having separate register callbacks, since then you could check the function arguments here, which I think is a good idea because then you can force specific names for the callback args. You might find `check_function2()` helpful: https://github.com/r-lib/httr2/blob/main/R/utils.R#L179-L235

Reply:
I implemented individual callback managers in 881cd2d8ff6b75f19c0eb1622c45ce5959a819d3 (or see [all changes in this second round](https://github.com/tidyverse/ellmer/pull/493/commits/881cd2d8ff6b75f19c0eb1622c45ce5959a819d3))